src/azure-cli/azure/cli/command_modules/resource/custom.py

```diff
@@ -2726,6 +2728,9 @@ def resolve_api_version(rcf, resource_provider_namespace, parent_resource_path,
         if not rt:
             raise IncorrectUsageError('Resource type {} not found.'.format(resource_type_str))
         if len(rt) == 1 and rt[0].api_versions:
+            # If latest_version is true, the last api-version will be taken regardless of whether it is "preview" or not
+            if latest_version:
+                return rt[0].api_versions[0]
             npv = [v for v in rt[0].api_versions if 'preview' not in v.lower()]
             return npv[0] if npv else rt[0].api_versions[0]
```

Comment:
Let's add some comments here. The previous code is hard to understand.

Reply:
OK~